packages/cli/src/external-secrets.ee/providers/aws-secrets-manager.ts

```diff
@@ -134,8 +171,60 @@ export class AwsSecretsManager implements SecretsProvider {
 	}
 
 	private assertAuthType(context: AwsSecretsManagerContext) {
-		if (context.settings.authMethod === 'iamUser') return;
+		const { authMethod } = context.settings;
+		if (authMethod === 'iamUser' || authMethod === 'autoDetect') return;
+		throw new UnknownAuthTypeError(authMethod);
+	}
+
+	private async fetchAllSecretsNames() {
+		const names: string[] = [];
+		let nextToken: string | undefined;
+
+		do {
+			const response = await this.client.listSecrets({
+				NextToken: nextToken,
+			});
 
-		throw new UnknownAuthTypeError(context.settings.authMethod);
+			if (response.SecretList) {
+				names.push(...response.SecretList.filter((s) => s.Name).map((s) => s.Name!));
```

Comment:
`Name` is mistyped or why are we ignoring the optionality?

Reply:
Oh you can use a predicate, e.g. `const numArr = [1, 2, "3"].filter((n): n is number => typeof n === "number")`

Reply:
I have another external-secrets refactor/cleanup PR in the pipeline. I'll address these there 🙏🏽